Sources/Datadog/CrashReporting/DDCrashReportingPluginType.swift

```diff
@@ -10,14 +10,140 @@ import Foundation
 /// Crash Report format supported by Datadog SDK.
 @objc
 public class DDCrashReport: NSObject {
+    public struct Thread: Encodable {
+        /// The name of the thread, e.g. `"Thread 0"`
+        internal let name: String
+        /// Unsymbolicated stack trace of the crash.
+        internal let stack: String
+        /// If the thread was halted.
+        internal let crashed: Bool
+        /// Thread state (CPU registers dump), only available for halted thread.
+        internal let state: String?
+
+        public init(
+            name: String,
+            stack: String,
+            crashed: Bool,
+            state: String?
+        ) {
+            self.name = name
+            self.stack = stack
+            self.crashed = crashed
+            self.state = state
+        }
+
+        // MARK: - Encoding
+
+        enum CodingKeys: String, CodingKey {
+            case name = "name"
+            case stack = "stack"
+            case crashed = "crashed"
+            case state = "state"
+        }
+    }
+
+    public struct BinaryImage: Encodable {
+        internal let libraryName: String
+        internal let uuid: String
+        internal let architecture: String
+        internal let isSystemLibrary: Bool
+        internal let loadAddress: String
+        internal let maxAddress: String
+
+        public init(
+            libraryName: String,
+            uuid: String,
+            architecture: String,
+            isSystemLibrary: Bool,
+            loadAddress: String,
+            maxAddress: String
+        ) {
+            self.libraryName = libraryName
+            self.uuid = uuid
+            self.architecture = architecture
+            self.isSystemLibrary = isSystemLibrary
+            self.loadAddress = loadAddress
+            self.maxAddress = maxAddress
+        }
+
+        // MARK: - Encoding
+
+        enum CodingKeys: String, CodingKey {
+            case libraryName = "name"
+            case uuid = "uuid"
+            case architecture = "arch"
+            case isSystemLibrary = "is_system"
+            case loadAddress = "load_address"
+            case maxAddress = "max_address"
+        }
+    }
+
+    /// Meta information about the process.
+    /// Ref.: https://developer.apple.com/documentation/xcode/examining-the-fields-in-a-crash-report
+    public struct Meta: Encodable {
+        /// A client-generated 16-byte UUID of the incident.
+        internal let incidentIdentifier: String?
+        /// The name of the crashed process.
+        internal let processName: String?
+        /// Parent process information.
+        internal let parentProcess: String?
+        /// The location of the executable on disk.
+        internal let path: String?
```

Comment:
Just FMI what is exactly this executable ?

Reply:
This is the location of the app's executable - it's main binary. An example value of this for `SimpleApp` would be:
```
/private/var/containers/Bundle/Application/10C61F35-5173-4AD9-8E3A-1C6C73478448/SimpleApp.app/SimpleApp
```